examples/acnn.py

```diff
@@ -0,0 +1,256 @@
+      val_metric_to_minimize = -1 * np.mean(val_metrics)
+    else:
+      val_metric_to_minimize = np.mean(val_metrics)
+
+    results.append((save_path, val_metric_to_minimize, val_metrics,
+                    test_metrics))
+
+    return val_metric_to_minimize
+
+  fmin(
+      objective,
+      candidate_hypers,
+      algo=tpe.suggest,
+      max_evals=args['num_trials'])
+  results.sort(key=lambda tup: tup[1])
+  best_trial_path, _, best_val_metrics, best_test_metrics = results[0]
+
+  copyfile(best_trial_path + '/configure.json',
+           args['result_path'] + '/configure.json')
+  copyfile(best_trial_path + '/eval.txt', args['result_path'] + '/eval.txt')
+
+  return best_val_metrics, best_test_metrics
+
+
+if __name__ == '__main__':
+  import argparse
+
+  from utils import mkdir_p
+
+  parser = argparse.ArgumentParser('Examples for MoleculeNet with ACNN')
+  parser.add_argument(
+      '-d',
+      '--dataset',
+      choices=['PDBbind'],
+      default='PDBbind',
+      help='Dataset to use')
+  parser.add_argument(
+      '-m',
+      '--model',
+      choices=['ACNN'],
+      default='ACNN',
+      help=
+      'Options include 1) Atomic Convolutional Neural Network (ACNN) (default: ACNN)'
+  )
+  parser.add_argument(
+      '-f',
+      '--featurizer',
+      choices=['AC'],
+      default='AC',
+      help='Options include 1) Atomic Convolution (AC) (default: AC)')
+  parser.add_argument(
+      '-p',
+      '--result-path',
+      type=str,
+      default='results',
+      help='Path to save training results (default: results)')
+  parser.add_argument(
+      '-r',
+      '--num-runs',
+      type=int,
+      default=3,
+      help='Number of runs for each hyperparameter configuration (default: 3)')
+  parser.add_argument(
+      '-pa',
+      '--patience',
+      type=int,
+      default=30,
+      help='Number of epochs to wait before early stop if validation performance '
+      'stops getting improved (default: 30)')
+  parser.add_argument(
+      '-hs',
+      '--hyper-search',
+      action='store_true',
+      help='Whether to perform hyperparameter search '
+      'or use the default configuration. (default: False)')
+  parser.add_argument(
+      '-nt',
+      '--num-trials',
+      type=int,
+      default=16,
+      help='Number of trials for hyperparameter search (default: 16)')
+  parser.add_argument(
+      '-me',
+      '--metric',
+      type=str,
+      choices=['rmse', 'r2'],
+      default='rmse',
+      help=
+      'Validation metric to optimize. Options inclue 1) rmse and 2) r2 (default: rmse)'
+  )
+  args = parser.parse_args().__dict__
+
+  mkdir_p(args['result_path'])
+  if args['hyper_search']:
+    print('Start hyperparameter search with Bayesian '
+          'optimization for {:d} trials'.format(args['num_trials']))
+    val_metrics, test_metrics = bayesian_optimization(args)
+  else:
+    print('Use the manually specified hyperparameters')
+    with open('configures/{}_{}/{}.json'.format(
```

Comment:
This file should also consider the metric to use.

Reply:
Fixed